cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -242,46 +242,54 @@
 ;; `{i,b}64` and smaller.
 
 ;; And two registers.
-(rule 0 (lower (has_type (fits_in_64 ty) (band x y)))
+(rule 0 (lower (has_type ty (band x y)))
+      (if (ty_int_ref_scalar_64 ty))
       (x64_and ty x y))
 
 ;; And with a memory operand.
 
-(rule 1 (lower (has_type (fits_in_64 ty)
-                       (band x (sinkable_load y))))
+(rule 1 (lower (has_type ty (band x (sinkable_load y))))
+      (if (ty_int_ref_scalar_64 ty))
       (x64_and ty x
                (sink_load_to_gpr_mem_imm y)))
 
-(rule 2 (lower (has_type (fits_in_64 ty)
-                       (band (sinkable_load x) y)))
+(rule 2 (lower (has_type ty (band (sinkable_load x) y)))
+      (if (ty_int_ref_scalar_64 ty))
       (x64_and ty
                y
                (sink_load_to_gpr_mem_imm x)))
 
 ;; And with an immediate.
 
-(rule 3 (lower (has_type (fits_in_64 ty)
-                       (band x (simm32_from_value y))))
+(rule 3 (lower (has_type ty (band x (simm32_from_value y))))
+      (if (ty_int_ref_scalar_64 ty))
       (x64_and ty x y))
 
-(rule 4 (lower (has_type (fits_in_64 ty)
-                       (band (simm32_from_value x) y)))
+(rule 4 (lower (has_type ty (band (simm32_from_value x) y)))
+      (if (ty_int_ref_scalar_64 ty))
       (x64_and ty y x))
 
+;; f32 and f64
+
+(rule 5 (lower (has_type (ty_scalar_float ty) (band x y)))
```

Comment:
We should probably use the same `if` syntax here for consistency...

Reply:
```
  src\isa\x64\lower.isle:275:3: type error: Used non-pure constructor 'ty_scalar_float' in pure expression context
  src\isa\x64\lower.isle:337:3: type error: Used non-pure constructor 'ty_scalar_float' in pure expression context
  ```
  
  ```
(decl ty_scalar_float (Type) Type)
(extern extractor ty_scalar_float ty_scalar_float)
```

I don't think this is possible now. It should be separate to other patch.